app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = Settings.valid_email_regex
+  before_save { email.downcase! }
+  validates :name, presence: true, length: {maximum: Settings.max_name}
+  validates :email, presence: true, length: {maximum: Settings.max_email},
+                    format: {with: VALID_EMAIL_REGEX},
+                    uniqueness: true
+  validates :password, presence: true, length: {minimum: Settings.min_password}
+  has_secure_password
```

Comment:
dòng này để làm gì e ?

Reply:
Khi được include trong 1 model, method này thêm chức năng sau:
• Khả năng lưu thuộc tính password_digest được băm an toàn vào cơ sở dữ liệu
• Một cặp thuộc tính (password and password_confirmation) xuất hiện khi tạo đối tượng và xác thực yêu cầu chúng khớp với nhau
• Phương thức xác thực trả về người dùng khi mật khẩu là đúng